translation/src/main/java/com/google/cloud/trace/zipkin/translation/TraceTranslator.java

```diff
@@ -89,15 +89,13 @@ private void finishTrace(Trace.Builder traceBuilder, Collection<Trace> converted
     }
   }
 
-  private String convertTraceId(long zipkinTraceId) {
-    // Stackdriver trace ID's are 128 bits = 16 bytes * 8
-    ByteBuffer idBuffer = ByteBuffer.allocate(16);
-    idBuffer.putLong(zipkinTraceId);
-    idBuffer.putLong(zipkinTraceId);
-    StringBuilder idBuilder = new StringBuilder();
-    for (byte b : idBuffer.array()) {
-      idBuilder.append(String.format("%02x", b));
-    }
-    return idBuilder.toString();
+  // Stackdriver trace ID's are 128 bits = 16 bytes * 8
+  private String convertTraceId(Span zipkinSpan) {
+    // zipkin conditionally encodes its id as 16 chars if the high bits are zero
+    // This unconditionally encodes as 32 chars as Stackdriver trace IDs are fixed length.
+    char[] result = new char[32];
+    Util.writeHexLong(result, 0, zipkinSpan.traceIdHigh);
```

Comment:
incidentally, we can always change this later.. I'll do this.. will revert to the original impl, but add a comment on why we are duplicating. Later it can be followed-up as necessary.